### ISSUE TITLE
Animation Callback affecting animation performance

### ISSUE THREAD

**Issue by Numel2020** [angular/angular#12732]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

So I have constructed a simple route animation from one component to the next that fades in one component and reveals the other. I have implemented simple animation callbacks which will fire when the *User Component* animation starts and when it also ends.
The callbacks work has expected and report themselves in the console when they are fired.
but there implementation causes the animation to have a delay on its start which is not what I am expecting. The only element that I can see that might be causing this is this piece of code:

     <div [@routeAnimation]="show"
     (@routeAnimation.start)="onStart($event)"
     (@routeAnimation.done)="onDone($event)">

**Expected behavior**

so the actually behaviour is to simply remove the routeAnimation prefixes with reference to the functions that it calls and you will notice seamless transition between one component to the next

**Minimal reproduction of the problem with instructions**

Please follow this link http://plnkr.co/edit/d1UQcN?p=preview

**What is the motivation / use case for changing the behavior?**

The method of callback should not impact on performance of the animation, where you are wanting to make callbacks on every animation this effect could be problematic

**Please tell us about your environment:**

PhpStorm 2016.2
OSx el capitain

* **Angular version:** 2.0.X

Angular 2.1.0

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

Chrome 54.0.2840.71 (64-bit)
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
typescript 2.0.3

* **Node (for AoT issues):** `node --version` =   
npm 3.10.4

**Comment by matsko** [comment]
hey @Numel2020 I believe this was greatly improved with https://github.com/angular/angular/commit/8395f0e138c3e9acd24f947fe9f077df55bee130.

This was also released in 2.3.1. Please reopen if it's still problematic.

**Comment by Numel2020** [comment]
hello @matsko 

Just reinstalled the CLI on my machine to the latest version, and checked my plunker, the update does not seem to have fixed the issue with the callback inclusion.
1. with the inclusion of this code
`<div [@routeAnimation]="show"
 (@routeAnimation.start)="onStart($event)"
 (@routeAnimation.done)="onDone($event)">`

I always experience a slight delay on the re-entry of the first div from the last one.

2. Now remove the callbacks from the div
`<div>`

The animation plays smooth without delay. 
[EXAMPLE](http://plnkr.co/edit/d1UQcN?p=preview)

Please have a look and tell me if I have made an error.

many thanks
